assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentEventObserver.java

```diff
@@ -0,0 +1,116 @@
+
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class AssignmentEventObserver implements Observer {
+
+    @Setter private AssignmentService assignmentService;
+    @Setter private EventTrackingService eventTrackingService;
+    @Setter private GradebookService gradebookService;
+
+    public void init() {
+        eventTrackingService.addLocalObserver(this);
+    }
+
+    public void destroy() {
+        eventTrackingService.deleteObserver(this);
+    }
+
+    @Override
+    public void update(Observable o, Object arg) {
+        if (arg instanceof Event) {
+            Event event = (Event) arg;
+            if (event.getModify() && StringUtils.isNoneBlank(event.getEvent(), event.getContext())) {
+                switch (event.getEvent()) {
+                    case "gradebook.updateItemScore": // grade updated in gradebook lets attempt to update the submission
+                        String[] parts = StringUtils.split(event.getResource(), '/');
+                        if (parts.length >= 5) {
+                            final String source = parts[0];
+                            final String gradebookId = parts[1];
+                            final String itemId = parts[2];
+                            final String studentId = parts[3];
+                            final String score = parts[4];
+                            log.debug("Updating score for user {} for item {} with score {} in gradebook {} by {}", studentId, itemId, score, gradebookId, source);
+                            if ("gradebookng".equals(source)) {
+                                Optional<Assignment> assignment = Optional.empty();
+                                Optional<AssignmentSubmission> submission = Optional.empty();
+                                // Assignments stores the gradebook item name and not the id :(, so we need to look it up
+                                try {
+                                    org.sakaiproject.service.gradebook.shared.Assignment gradebookAssignment = gradebookService.getAssignmentByNameOrId(event.getContext(), itemId);
+                                    assignment = Optional.ofNullable(assignmentService.getAssignmentForGradebookLink(event.getContext(), gradebookAssignment.getName()));
+                                    if (assignment.isPresent()) {
+                                        Assignment a = assignment.get();
+                                        submission = Optional.ofNullable(assignmentService.getSubmission(a.getId(), studentId));
+                                        if (submission.isPresent()) {
+                                            AssignmentSubmission s = submission.get();
+                                            final String grade;
+                                            if (Assignment.GradeType.SCORE_GRADE_TYPE.equals(a.getTypeOfGrade())) {
+                                                int dec = (int) Math.log10(a.getScaleFactor());
```

Comment:
I don't know it it's a good idea to be replicating that scaling logic from assignments in here like this. This should be in a common utils file, or something like that, imo.

Reply:
The reason for it, is that logic is in the AssignmentAction and I didn't want to add it to the service because it's a really bad.